pkg/ipcache/cidr.go

```diff
@@ -38,6 +38,22 @@ func AllocateCIDRs(impl Implementation, prefixes []*net.IPNet) error {
 		return err
 	}
 
+	// insert these locally. FromAgentLocal cannot be overwritten.
+	// stolen from pkg/ipcache/kvstore.go:158, UpsertIPToKVStore
+	for idx, prefix := range prefixes {
+		ipIDPair := identity.IPIdentityPair{
+			IP:       prefix.IP,
+			Mask:     prefix.Mask,
+			ID:       prefixIdentities[idx].ID,
+			Metadata: AddressSpace, // XXX: Should we associate more metadata?
+		}
+
+		IPIdentityCache.Upsert(ipIDPair.PrefixString(), ipIDPair.HostIP, Identity{
+			ID:     ipIDPair.ID,
+			Source: FromAgentLocal,
+		})
+	}
+
```

Comment:
In the error paths a few lines below I think we'll need to walk back these changes, otherwise in error cases we'll end up with entries stuck in the IPCache forever.

Also, what would happen if multiple rules are imported with the same CIDR? The IPIdentityCache doesn't inherently have any reference tracking for these CIDRs right? (Only prefix length tracking).

Reply:
oh, you're totally right. I thought the refcount code in `kvReferenceCounter` was in here... ugh...